packages/types/src/codec/EnumType.ts

```diff
@@ -127,6 +127,30 @@ export default class Enum extends Base<Codec> implements Codec {
     return class extends Enum {
       constructor (value?: any, index?: number) {
         super(Types, value, index);
+
+        Object.keys(this._def).forEach((_key) => {
+          const askey = `as${_key}`;
+          const iskey = `is${_key}`;
+
+          // do not clobber existing properties on the object
+          if (isUndefined((this as any)[iskey])) {
+            Object.defineProperty(this, iskey, {
+              enumerable: true,
+              get: () => this.type === _key
```

Comment:
Need to add an explicit test for this - I'm not convinced it is working as intended for basic enums. (Generally the `toNumber() === x` approach is used there)